cowbird/services/impl/geoserver.py

```diff
@@ -26,3 +31,42 @@ def permission_created(self, permission):
 
     def permission_deleted(self, permission):
         raise NotImplementedError
+
+    def create_workspace(self, name):
+        # type (Geoserver, str) -> int
+        """
+        Create a new Geoserver workspace.
+
+        @param self: Geoserver instance
+        @param name: Workspace name
+        @return: Workspace id
+        """
+        # TODO
+        return 1
+
+    def create_datastore(self, workspace_id, name):
+        # type (Geoserver, int, str) -> int
+        """
+        Create a new Geoserver workspace.
+
+        @param self: Geoserver instance
+        @param workspace_id: Workspace id where the datastore must be created
+        @param name: Datastore name
+        @return: Datastore id
+        """
+        # TODO
+        return 1
+
+
+@shared_task(bind=True, base=RequestTask)
+def create_workspace(self, name):
+    # type (str) -> int
+    # Avoid any actual logic in celery task handler, only task related stuff should be done here
+    return ServiceFactory().get_service("Geoserver").create_workspace(name)
+
+
+@shared_task(bind=True, base=RequestTask)
+def create_datastore(self, workspace_id, name):
+    # type (int, str) -> int
+    # Avoid any actual logic in celery task handler, only task related stuff should be done here
+    return ServiceFactory().get_service("Geoserver").create_datastore(workspace_id, name)
```

Comment:
I'm curious. Can't these decorators be applied directly onto (static) methods within `Geoserver` class?

Reply:
They could, but here I show a stub of what should be done. In the end the function will need the class instance to get the server config for example.
And since the shared_task is also bind to the celery task, we cannot mix and match the celery task self and the geoserver self. I tried something, but in the end it was uglier.